go/pkg/filemetadata/cache_test.go

```diff
@@ -112,3 +114,56 @@ func TestLoadAfterChangeWithoutValidation(t *testing.T) {
 		t.Errorf("Cache has wrong num of CacheMisses, want 1, got %v", c.GetCacheMisses())
 	}
 }
+
+func TestWithDirContent(t *testing.T) {
```

Comment:
Do you want to write a test where you call `c.Get()`, add a file, call `c.Get()` again and ensure that the second file does NOT show up? This would confirm the cache actually works.